services/supabase/gitauto_manager.py

```diff
@@ -137,6 +154,16 @@ def create_user_request(
             )
             .execute()
         )
+        
+        if not str(email).lower().endswith("@users.noreply.github.com"):
+            response: APIResponse = self.client.table("users").select("email").eq("user_id", user_id).execute()
+
+            if response.data and len(response.data[0]) > 0:
```

Comment:
1. We should separate user creation/update from create_user_request
2. We should create a user or update it not only updating it.

Reply:
Let's talk about this